packages/react-components/react-tree/src/components/Tree/Tree.types.ts

```diff
@@ -39,6 +45,24 @@ export type TreeOpenChangeData = {
 
 export type TreeOpenChangeEvent = TreeOpenChangeData['event'];
 
+export type TreeCheckedChangeData = {
+  value: TreeItemValue;
+  target: HTMLElement;
+  event: React.ChangeEvent<HTMLElement>;
+  type: 'Change';
+} & (
+  | {
+      selectionMode: 'multiselect';
```

Comment:
why return the selectionMode back to the user?

Reply:
because `checked` value might change depending on the selection mode

1. if it's `multiselect` then `checked` is `boolean | mixed`
2. if it's `single` then `checked` is `boolean`

Reply:
one other possibility is having a `MultiSelectChange` and a `SingleSelectChange` `type` instead of returning the `selectionMode`

Reply:
ah I see, that's an interesting way to handle the `mixed` value